lib/Controller/UsersController.php

```diff
@@ -469,6 +516,211 @@ public function create($username, $password, array $groups= [], $email='') {
+			], 'guest'
+		);
+	}
+
+	/**
+	 * @param string $token
+	 * @param string $userId
+	 * @return null
+	 * @throws InvalidUserTokenException
+	 * @throws UserTokenExpiredException
+	 * @throws UserTokenMismatchException
+	 */
+	private function checkPasswordSetToken($token, $userId) {
+		$user = $this->userManager->get($userId);
+
+		$splittedToken = \explode(':', $this->config->getUserValue($userId, 'owncloud', 'lostpassword', null));
+		if (\count($splittedToken) !== 2) {
+			$this->config->deleteUserValue($userId, 'owncloud', 'lostpassword');
+			throw new InvalidUserTokenException($this->l10n->t('The token provided is invalid.'), UserTokenException::TOKEN_AND_TIME_INVALID);
+		}
+
+		if ($splittedToken[0] < ($this->timeFactory->getTime() - (int)$this->config->getAppValue('user_management', 'token_expire_time', '0')) ||
+			$user->getLastLogin() > $splittedToken[0]) {
+			$this->config->deleteUserValue($userId, 'owncloud', 'lostpassword');
+			throw new UserTokenExpiredException($this->l10n->t('The token provided had expired.'), UserTokenException::TOKEN_TIME_EXPIRED);
+		}
+
+		if (!\hash_equals($splittedToken[1], $token)) {
+			$this->config->deleteUserValue($userId, 'owncloud', 'lostpassword');
+			throw new UserTokenMismatchException($this->l10n->t('The token provided is invalid.'), UserTokenException::TOKEN_MISMACTH);
+		}
+	}
+
+	/**
+	 * @PublicPage
+	 * @NoCSRFRequired
+	 * @NoAdminRequired
+	 * @NoSubadminRequired
+	 * @param $userId
+	 */
+	public function resendToken($userId) {
+		$user = $this->userManager->get($userId);
+
+		if ($user === null) {
+			return new TemplateResponse(
+				'core', 'error',
+				[
+					"errors" => [["error" => $this->l10n->t('User: %s does not exist', [$userId])]]
+				],
+				'guest'
+			);
+		}
+
+		if ($user->getEMailAddress() === null) {
+			return new TemplateResponse(
+				'core', 'error',
+				[
+					"errors" => [["error" => $this->l10n->t('Email address not set for: %s', [$userId])]]
+				],
+				'guest'
+			);
+		}
+
+		try {
+			$this->generateTokenAndSendMail($user->getUID(), $user->getEMailAddress());
+		} catch (\Exception $e) {
+			$this->log->error("Can't send new user mail to " . $user->getEMailAddress() . ": " . $e->getMessage(), ['app' => 'settings']);
+			return new TemplateResponse(
+				'core', 'error',
+				[
+					"errors" => [[
+						"error" => $this->l10n->t('Can\'t send email to %s for the new user. Contact your administrator.', [$user->getEMailAddress()])]]
+				], 'guest'
+			);
+		}
+
+		return new TemplateResponse(
+			'user_management', 'new_user/tokensendnotify', [], 'guest'
+		);
+	}
+
+	/**
+	 * @PublicPage
+	 * @NoAdminRequired
+	 * @NoSubadminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @param $token
+	 * @param $userId
+	 * @param $password
+	 * @return JSONResponse
+	 */
+	public function setPassword($token, $userId, $password) {
+		$user = $this->userManager->get($userId);
+
+		if ($user === null) {
+			return new JSONResponse(
```

Comment:
same as https://github.com/owncloud/user_management/pull/48/files#r212668911, please check all error messages generated for public pages

Reply:
Made the message more generic.